osu.Game.Tests/Visual/TestCaseEditorSelectionLayer.cs

```diff
@@ -68,6 +69,7 @@ private void load(OsuGameBase osuGame)
             var clock = new DecoupleableInterpolatingFramedClock { IsCoupled = false };
             dependencies.CacheAs<IAdjustableClock>(clock);
             dependencies.CacheAs<IFrameBasedClock>(clock);
+            dependencies.Cache(new BindableBeatDivisor());

```

Comment:
rather than introducing cross-dependencies like this, just set a default value to the bindable on the components being tested and only bind externals if not-null.

Reply:
You're going to have to define what "cross-dependency" means.

Idk, I feel like that's going to bite us (or someone else reading the code) in the arse when they go "permitNulls is true, so I can just pass a null game and clocks".

Reply:
The idea behind DI/bindables is that when testing, you don't need to provide every component.

If unavailable, the components will still be testable on their own. You should only be providing dependencies if you require them to do further testing by altering/querying them. tests should be as restricted in scope as possible.

Reply:
It's also the responsibility of testcases to put components in a valid environment, and sometimes that requires also having clocks, the game/osucolour, etc, like most of the editor components currently tested. Sometimes that also requires that these components have a beat snap divisor to look onto.

Personally I think it's actually good that the bindable is defined here in this test case, depending on how we want to continue doing this test in the future we may change the value - it does support seeking after all.

I'm just not sure it's worth it, or is even an issue at all.

Reply:
We may want to consider moving the bindable to ctor in this case... since it becomes a required dependency.